tests/pipelines/regression/v1/test_batch_scoring_dag.py

```diff
@@ -0,0 +1,64 @@
+import pandas as pd
+import pathlib
+import pytest
+import shutil
+
+import mlflow
+from mlflow.pipelines.regression.v1.pipeline import RegressionPipeline
+
+# pylint: disable=unused-import
+from tests.pipelines.helper_functions import (
+    enter_pipeline_example_directory,
+    get_execution_directory_location,
+    PIPELINE_EXAMPLE_PATH_FROM_MLFLOW_ROOT,
+    chdir,
+)  # pylint: enable=unused-import
+
+_STEP_NAMES = ["ingest_scoring", "predict"]
+
+
+# Because the Batch Scoring DAG takes a lot of time to run (around 5 minutes), we run the entire
```

Comment:
What's the reason it takes so long to run? Data size? Spark start-up? Can we speed it up to make it finish within 1 min?

Reply:
I think it is the environment restoration in the Spark UDF. When the predict step is run with the `p.run` API, I have no been able to find a way to patch the env_manager to be `"local"` to speed up the process. Thus it is quite slow.

Other reasons might be that it runs the entire training DAG + starts up spark.

Reply:
I see two ways to fix this: one is to expose the env_manager in the step config. Two is to speed up environment restoration by detecting if the inference environment is the same as the training environment, and not materialize a new environment if that is the case.

Reply:
I've also started a slack thread on this in the OSS channel. I'm currently not leaning towards exposing the env_manager in the step config, especially since we plan to deprecate the `local` env_manager in MLflow 2.0.

Reply:
SG. Let's see how this could be better solved.